src/scripts/theme-toggle-persistence.js

```diff
@@ -11,28 +11,28 @@
  */
 
 export function initThemeTogglePersistence() {
+
+  const themeToggleWrapper = document.getElementById('theme-toggle');
+  const themeToggleLabel = themeToggleWrapper?.querySelector('label');
+  const themeToggleCheckbox = themeToggleWrapper?.querySelector('input[type="checkbox"]');
+
+  if (!themeToggleWrapper || !themeToggleLabel || !themeToggleCheckbox) return;
+
+  const root = document.documentElement;
+
+
   const userThemePreference = !!(
     window.matchMedia &&
     window.matchMedia('(prefers-color-scheme: dark)').matches
   );
-  const toggle = document.getElementById('theme-toggle');
-  if (!toggle) return;
-
-  const root = document.documentElement;
 
   const storageKey = 'salud-theme';
   let didWarn = false;
 
   const syncAria = (isDark) => {
-    const themeLabel = isDark ? 'Theme: dark' : 'Theme: light';
-    toggle.setAttribute('aria-checked', String(isDark));
-    toggle.setAttribute('aria-pressed', String(isDark));
-    toggle.setAttribute('aria-label', themeLabel);
-    const labelEl = document.querySelector('label[for="theme-toggle"].theme-toggle');
-    if (labelEl) {
-      labelEl.setAttribute('aria-pressed', String(isDark));
-      labelEl.setAttribute('aria-label', themeLabel);
-    }
+    const ariaLabel = isDark ? 'Theme: dark' : 'Theme: light';
+    themeToggleLabel.setAttribute('aria-pressed', String(isDark));
```

Comment:
If you are using a custom visual switch (label is the interactive element with role="switch" / role="button"):
Make the label the interactive element (focusable) and give it an appropriate role (e.g., role="switch") and then use aria-checked (not aria-pressed for a switch) on that element: themeToggleLabel.setAttribute('role', 'switch'); themeToggleLabel.setAttribute('aria-checked', String(isDark)); themeToggleLabel.setAttribute('aria-label', ariaLabel);